test/compilable/b18242.d

```diff
@@ -0,0 +1,15 @@
+// REQUIRED_ARGS: -c
```

Comment:
If you add `DFLAGS:` (without anything following it) it will prevent all the default flags for importing the runtime, etc.. from being added implicitly to compiler invocation.  Then you shouldn't need to declare the `TypeInfo` stuff and have to deal with all object mismatch errors.

See #7845

Reply:
Sorry, Ignore the comment above.  It appears `TypeInfo` is required because `Throwable` has been declared.  This should get you passed the 32-bit tests:

```D
module object;

class Object { }

class Throwable { }

class TypeInfo { }
class TypeInfo_Class : TypeInfo 
{ 
    version(D_LP64)
    {
        ubyte[136] _x; 
    }
    else
    {
        ubyte[68] _x;
    }
}

int _d_run_main() 
{
    try { } catch(Throwable e) { return 1; }
    return 0;
}
```

Reply:
Thanks for the hint, I always forget about 32bit platforms.